appinventor/appengine/src/com/google/appinventor/client/widgets/SearchBox.java

```diff
@@ -0,0 +1,222 @@
+package com.google.appinventor.client.widgets;
+
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.event.dom.client.KeyDownEvent;
+import com.google.gwt.event.dom.client.KeyDownHandler;
+import com.google.gwt.event.dom.client.KeyUpEvent;
+import com.google.gwt.event.dom.client.KeyUpHandler;
+import com.google.gwt.event.dom.client.KeyPressEvent;
+import com.google.gwt.event.dom.client.KeyPressHandler;
+import com.google.gwt.user.client.Command;
+import com.google.gwt.event.dom.client.KeyCodes;
+import com.google.appinventor.client.editor.youngandroid.BlocklyPanel;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.SuggestBox;
+import com.google.gwt.user.client.ui.MultiWordSuggestOracle;
+import com.google.appinventor.client.widgets.TextButton;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.user.client.ui.Image;
+import com.google.appinventor.client.output.OdeLog;
+import com.google.gwt.user.client.ui.CheckBox;
+import com.google.gwt.user.client.ui.ListBox;
+import static com.google.appinventor.client.Ode.MESSAGES;
+import com.google.gwt.event.dom.client.ChangeEvent;
+import com.google.gwt.event.dom.client.ChangeHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.script.*;
+import java.lang.Double.*;
+
+import java.lang.String;
+
+public class SearchBox extends Composite {
+
+    private BlocklyPanel blockArea;
+    private SuggestBox searchBox;
+    private CheckBox advancedCheckBox;
+    private ListBox filterBox;
+    private TextButton upButton, downButton,clearButton;
+    private char upArrow = '\u2191';
+    private char downArrow = '\u2193';
+    public int flag;
+
+    public SearchBox(BlocklyPanel panel) {
+        
+
+        /* Panel*/
+        blockArea = panel;
```

Comment:
Indentation here is still not right. 2 spaces for all new lines and 4 for a continuation line.